diff --git a/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts b/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts
index 5f248fb01bca..43b7659f4254 100644
--- a/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts
+++ b/arch/arm/boot/dts/bcm2710-rpi-3-b-plus.dts
@@ -20,6 +20,13 @@ aliases {
 		serial1 = &uart0;
 		mmc1 = &mmcnr;
 	};
+
+	 mychardev: mychardev@0 {
+        compatible = "akshay,mychardev";
+        my-gpio = <&gpio 17 GPIO_ACTIVE_HIGH>;
+        timer-interval-ms = <10000>;
+        status = "okay";
+    };
 };
 
 &gpio {
